fil-ir/src/from_ast/sig_map.rs

```diff
@@ -1,49 +1,44 @@
+use super::BuildRes;
 use crate as ir;
 use fil_ast as ast;
 use fil_utils::{self as utils, Error, Id};
 use itertools::Itertools;
 use std::collections::HashMap;
 
-use super::BuildRes;
-
 #[derive(Clone)]
 /// The signature of component.
 ///
 /// A signature defines the ports which are added to the component instantiating
 /// the signature.
 pub struct Sig {
     pub idx: ir::CompIdx,
-    pub events: Vec<ir::EventIdx>,
-    pub inputs: Vec<ir::PortIdx>,
-    pub outputs: Vec<ir::PortIdx>,
+    /// Mapping from source representation of input port to its IR index
+    pub inputs: Vec<(ast::Loc<ast::PortDef>, ir::PortIdx)>,
```

Comment:
The previous encoding relied on `inputs` containing the input ports in the same order as `raw_inputs`. This makes that dependency explicit.